File: Server/public/assets/js/city.js
function capitalizeFirstLetterOnly (string) {
    return string.charAt(0).toUpperCase() + string.slice(1).toLowerCase();
}

// Used for registering a city into server database
function register() {
    $("#successfulMessage").addClass("hidden");
    $("#error").addClass("hidden");

    let name = $("#inputName").val();
    let region = $("#inputRegion").val();

    // Reverse geocoding use capital first letter, so to improve compatibility also save the fields in this format
    name = capitalizeFirstLetterOnly(name);
    region = capitalizeFirstLetterOnly(region);

    let json = JSON.stringify({name, region});
    console.log(json);

    $.ajax({
        type: "POST",
        url: "/v2/admin/cities/register",
        data: json,
        contentType: "application/json; charset=utf-8",
        success: function(data){
            $("#successfulMessage").removeClass("hidden");
        },
        error: function(errMsg) {
            $("#error").removeClass("hidden").html(errMsg.statusText);
        }
    });
}

File: Client/safestreets/lib/models/city.dart
class City {
  int _id;
  String _name;
  String _region;
  City(this._id);

  int get id => _id;
  String get name => _name;
  String get region => _region;

  City.map(dynamic obj) {
    this._id = obj["id"];
    this._name = obj["name"];
    this._region = obj["region"];
  }

  Map<String, dynamic> toMap() {
    var map = new Map<String, dynamic>();
    map["id"] = _id;
    map["name"] = _name;
    map["region"] = _region;
    return map;
  }


}